blueocean-rest-impl/src/test/java/io/jenkins/blueocean/util/HttpRequest.java

```diff
@@ -0,0 +1,225 @@
+package io.jenkins.blueocean.util;
+
+import com.damnhandy.uri.template.UriTemplate;
+import com.google.common.base.Preconditions;
+import io.jenkins.blueocean.commons.JsonConverter;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.fluent.Content;
+import org.apache.http.client.fluent.ContentResponseHandler;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.fluent.Response;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.log4j.Logger;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.jenkins.blueocean.util.HttpRequest.Method.GET;
+
+/**
+ * TODO: needs javadoc once API is final
+ * @author cliffmeyers
+ */
+public class HttpRequest {
+
+    private Logger logger = Logger.getLogger(getClass());
+    private static String JSON = "application/json";
+
+    enum Method { GET, POST, PUT, PATCH, DELETE };
+
+    final private Map<String, String> headers = new HashMap<>();
+    final private Map<String, Object> urlParts = new HashMap<>();
+    private Method method = GET;
+    private int expectedStatus = 200;
+    private String baseUrl = "";
+    private String requestUrl;
+    private String contentType;
+    private String requestBody;
+    private String username;
+    private String password;
+
+
+    private HttpRequest() {}
+
+    public static HttpRequest build(String baseUrl) {
+        Preconditions.checkState(StringUtils.isNotBlank(baseUrl), "baseUrl is required");
+        HttpRequest builder = new HttpRequest();
+        builder.baseUrl(baseUrl);
+        return builder;
+    }
+
+    public HttpRequest baseUrl(@Nonnull String baseUrl) {
+        this.baseUrl = baseUrl;
+        return this;
+    }
+
+    public HttpRequest GET(String url) {
```

Comment:
Might be easier API to expose methods (HTTP methods) as public static and then remove `build` method and move the code that was there in to `execute`? 

Something along the line of:
```
Get("http://localhost:8080/jenkins/blue/rest/organizations/).auth("","").expectedStatus(200).as(Map.class).execute();
```

Reply:
That's similar style as to Apache Fluent, however having the entry point me the static "method" methods require us to specify the absolute URL each time. I'd argue that in almost every test class, there is a baseUrl that is the same and specifying it in each test method (even with uri templates) clutter things up. e.g.

Get("__http://localhost:8080/jenkins/blue/rest__ /organizations/jenkins")
vs
Get("__{base}__ /organizations/jenkins") __.urlPart("base", baseUrl)__
vs just
Get("/organizations/jenkins")

In 5fcba06 I made a small change so entrypoint is either static `build()` or `build(String baseUrl)` methods, so if you prefer the first style you just need `build()` before the `Get(...)`. If you feel strongly we should go the other way let me know and we can discuss.